src/layout/core/graph_tree.rs

```diff
@@ -684,6 +715,33 @@ impl InnerTree {
             node_ix = parent_ix;
         }
     }
+
+    /// Gets the next sibling to focus on, assuming child_ix would be removed from parent_ix.
+    /// If there are no other siblings to focus on, parent_ix is returned.
+    ///
+    /// # Panics
+    /// This will panic if `child_ix` is not a child of `parent_ix`
+    pub fn next_sibling(&self, parent_ix: NodeIndex, child_ix: NodeIndex) -> NodeIndex {
+        let children = self.children_of(parent_ix);
+        let mut prev_index = None;
+        for (index, sibling_ix) in children.iter().enumerate() {
+            if child_ix == *sibling_ix {
+                prev_index = Some(index);
+            }
+        }
+        if prev_index.is_none() {
```

Comment:
All this work is useless isn't it? We could just as easily get the weight, and use that as the index since that's how the results of children are sorted anyways!